pkg/compactor/compactor.go

```diff
@@ -520,6 +523,24 @@ func (c *Compactor) compactUsers(ctx context.Context) {
 		level.Info(c.logger).Log("msg", "successfully compacted user blocks", "user", userID)
 	}
 
+	// Delete local files for skipped tenants, if there are any.
+	for _, userID := range skippedTenants {
```

Comment:
This will delete data for tenants which don't belong anymore to the shard, but will keep data on disk if a tenant is deleted or a resharding occured while this compactor replica was not running (but disk is persisted). WDYT?

Reply:
Resharding is not a problem I think, but deleted tenants are. I will fix that.

Reply:
I think this is resolved. Please reopen if not.

Reply:
I agree it's solved now 👍